File: server/model/todo.model.js
const mongoose = require('mongoose');
const db = require('../config/db');

const { Schema } = mongoose;

const todoSchema = new Schema({

    title: {
        type: String,
        required: true
    },

    description: {
        type: String,
        required: true
    },

    createdAt: {
        type: Date,

        default: () => new Date().toUTCString()
    }
},
    { toJSON: { virtuals: true }, id: false }
);
todoSchema.virtual('id').get(function () {
    return this._id.toHexString();
});

const todoModel = db.model('todo', todoSchema);

module.exports = todoModel;

File: client/lib/model/todo_model.dart
class Todo {
  String id;
  String title;
  String description;
  DateTime createdAt;

  Todo({
    required this.id,
    required this.title,
    required this.description,
    required this.createdAt,
  });

  factory Todo.fromJson(Map<String, dynamic> json) {
    return Todo(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}
